packages/react-storage/src/components/StorageBrowser/Views/LocationActionView/UploadControls.tsx

```diff
@@ -212,27 +196,39 @@ export const UploadControls = (): JSX.Element => {
     preventOverwrite: !overwrite,
   });
 
-  const [fileSelect, handleSelect] = useFileSelect(handleFileSelect);
+  const handleFileInput = React.useRef<HTMLInputElement>(null);
+  const handleFolderInput = React.useRef<HTMLInputElement>(null);
+
+  // Noticed that in Safari, the file picker was not registering the on change event
+  // unless I made sure that the useEffect for clicking the file input was only clicked once
+  const initialRun = React.useRef(false);
 
   let tableData = tasks.map((task) => {
-    const folder = task.data.webkitRelativePath.lastIndexOf('/') + 1;
+    const { webkitRelativePath } = task.data;
+
+    const folder =
+      webkitRelativePath?.length > 0
+        ? webkitRelativePath.slice(0, webkitRelativePath.lastIndexOf('/') + 1)
+        : '/';
```

Comment:
updated this cause I saw tests were throwing errors when trying to access `lastIndexOf` the `webkitRelativePath` property when it was undefined. Realized the original logic for this was wrong and so I updated it here